java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthVector.java

```diff
@@ -522,15 +530,44 @@ public void reallocDataBuffer() {
    * @throws OutOfMemoryException if the internal memory allocation fails
    */
   public void reallocValidityAndOffsetBuffers() {
-    offsetBuffer = reallocBufferHelper(offsetBuffer, true);
-    validityBuffer = reallocBufferHelper(validityBuffer, false);
+    int offsetBaseSize = Integer.max(offsetBuffer.capacity(), offsetAllocationSizeInBytes);
```

Comment:
I don't understand why this can't be generic just like the the original alloc with just valuecount *2

Reply:
thinking further, I think this can be simplified significantly if I make both the initial allocation and the reAlloc target a multiple-of-2 allocation. 

I'll need to fix a lot of tests (the allocation may be larger than what was asked for), but it's probably worth the simplicity.